streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -692,6 +699,7 @@ public boolean process(final long wallClockTime) {
                 return false;
             }
         }
+        timeCurrentIdlingStarted = Optional.empty();
```

Comment:
Why do we need this?

Reply:
My thought was if the record becomes not null after It was not processable in the previous iteration we would not call isProcessable. Is this case guaranteed not to happen? If so we don't need this

Reply:
We only set `record = partitionGroup.nextRecord(recordInfo, wallClockTime);` above, ie, only after the task is processable. Hence, the scenario you describe should not happen IMHO.

Reply:
Alright will remove then